massgov/mayflower
# Refactor button variants and classnames

## Description
Based on the updating of the global color variables and agreed upon button variants in our documentation, this pr refactors the button component classsnames - adding color variable modifiers to classnames and creating clear classname modifiers for primary, secondary, tertiary, and quaternary buttons based on usage.

Impact: Major Release.

## Related Issue / Ticket / Documentation

- Marked as WIP progress in documentation here: https://massds.gitbook.io/edit-mayflower-doc/components/atoms/buttons#classnames

## Steps to Test
* Key thing here is to confirm that this refactor will not negatively impact mass.gov @mrossi113 or @ygannett ?

Pattern Lab:
1. Button as Quaternary --> http://localhost:3000/?p=atoms-button-as-quaternary
2. Button as Tertiary --> http://localhost:3000/?p=atoms-button-as-tertiary
3. Button as Secondary --> http://localhost:3000/?p=atoms-button-as-secondary
4. Button as Secondary in Primary Alt Color --> http://localhost:3000/?p=atoms-button-as-secondary-primary-alt

React
1. npm install
2. npm start
3. Go to storybook and confirm below is possible.
![button-refactor](https://user-images.githubusercontent.com/9359261/45383250-b2ac4d80-b5d8-11e8-88c8-213ee41f77bb.gif)

#### Impacted Areas in Application
Anywhere using the button component, particularly the old button with outline which is refactored in this pr to secondary button.
